File: Serveur/interfaces/room.ts
import Document from 'mongoose'
import { IMessage } from './IMessage'

export interface IRoomDB extends Document  {
    name           :   string,
    owner          :   string,
    password       :   string,
    history        :   Array<IMessage>,
    users          :   Array<string>
}

export interface IRoom  {
    name           :   string,
    owner          :   string | undefined,
    password       :   string,
    history        :   Array<IMessage>,
    users          :   Array<string>,
    connectedUsers :   Array<string>
}

File: Client-Leger/app/src/main/java/com/zouatene/colorimage/model/chat/Room.kt
package com.zouatene.colorimage.model.chat

data class Room(
    var name: String,
    val owner: String,
    val password: String? = null,
    val history: List<Message>,
    val users: List<String>,
    val connectedUsers: List<String>,
    var unread: Int = 0
) : SectionedRoomListItem(TYPE_ROOM_ITEM)

data class RoomSection(
    val title: String,
) : SectionedRoomListItem(TYPE_ROOM_SECTION)

